Issue9

dongryeol-kim opened the pull request:
휴가 관리
기능: 사용자 휴가 신청 및 관리 기능을 제공합니다. 승인 절차를 통해 어드민이 휴가 여부를 검토할 수 있습니다.
게시글 관리 (Post Management)
게시글 작성 CRUD 작성
카테고리 관리 : 게시글의 종류에 따라 카테고리를 설정하여, 필요에 따라 공지사항, 자유 게시판 등의 카테고리로 구분합니다.
댓글 및 대댓글 작성:
댓글 기능을 제공하여 사용자 간의 의견을 주고받을 수 있으며, 최대 대댓글까지만 작성가능합니다.

dongryeol-kim committed:
ISSUE7
- 기능 : 먼저 주간/월간/일간을 구별하여 db에 insert 하면 db에 저장
  기능 : invitedemail 에 있는 email로 메일 보내기

dongryeol-kim committed:
ISSUE7
- 게시글 작성 (어드민 전용)
어드민은 전체 직원이 볼 수 있는 공지사항이나 기타 중요 게시글을 작성할 수 있습니다. 게시글을 만들지 않아서 어드민 AOP만 구현하여 USER 에서 insert를 가능하게 하거나 못하게 하는걸로 먼저 구현해놨습니다.
직급 설정
대표
본부장
실장
팀장
사원

dongryeol-kim committed:
ISSUE7
- 게시글 작성 (어드민 전용)
어드민은 전체 직원이 볼 수 있는 공지사항이나 기타 중요 게시글을 작성할 수 있습니다. 게시글을 만들지 않아서 어드민 AOP만 구현하여 USER 에서 insert를 가능하게 하거나 못하게 하는걸로 먼저 구현해놨습니다.
직급 설정
대표
본부장
실장
팀장
사원

dongryeol-kim committed:
issue6 - 코드리뷰 피드백 수정
- 유저 additionalInfo typehandler 추가하여 json 데이터 저장

dongryeol-kim committed:
issue9 - 코드리뷰 피드백 수정
- Article CRUD 동작 수정

dongryeol-kim committed:
issue9 - user테이블 id -> userNo 변경

dongryeol-kim committed:
issue9
 - user테이블 id -> userNo 변경
 - Vacation db 연동 구현
 - vacation 승인 및 거부 기능 추가

dongryeol-kim committed:
issue9
 - user테이블 id -> userNo 변경
 - Vacation db 연동 구현
 - vacation 승인 및 거부 기능 추가

dongryeol-kim committed:
issue9
 - user테이블 id -> userNo 변경
 - Vacation db 연동 구현
 - vacation 승인 및 거부 기능 추가

ccommit-dev commented on src/main/java/com/commit/collaboration_board_server/controller/ArticleController.java:
```diff
@@ -0,0 +1,46 @@
+        return ResponseEntity.ok(articleService.getAllArticles());
+    }
+
+    @PutMapping("/{id}")
```
put method와 patch 메서드 차이를 댓글로 달아주세요

ccommit-dev commented on src/main/java/com/commit/collaboration_board_server/service/ArticleService.java:
```diff
@@ -0,0 +1,39 @@
+
+@Service
+@RequiredArgsConstructor
+public class ArticleService {
```
service 로직에 예외처리를 추가하여 로깅되게 해주세요

ccommit-dev commented on src/main/java/com/commit/collaboration_board_server/model/Article.java:
```diff
@@ -0,0 +1,18 @@
+import lombok.Data;
+import java.time.LocalDateTime;
+
+@Data
```
@Data 어노테이션이 포함하는 기능 리스트를 댓글로 달아주세요

ccommit-dev commented on src/main/java/com/commit/collaboration_board_server/service/ArticleService.java:
```diff
@@ -0,0 +1,39 @@
+    private final ArticleMapper articleMapper;
+
+    public void createArticle(Article article) {
+        article.setViewCount(0);
```
객체지향 관점에서 @Setter를 사용하면 왜안좋은지 찾아서 이유 댓글달기

dongryeol-kim replied:
외부에서 객체의 상태나 내용, 데이터를 직접 바로 바꿀 수 있다는 장점이 있지만 객체지향적 관점에서는 객체의 세부정보를 보호하고 객체에 요청을 하면 얻을 수 있는 의미가 없어지게 됩니다.

ccommit-dev commented on src/main/java/com/commit/collaboration_board_server/controller/ArticleCategoryController.java:
```diff
@@ -0,0 +1,21 @@
+@RestController
+@RequestMapping("/article-categories")
+@RequiredArgsConstructor
+public class ArticleCategoryController {
```
Aritcle Controller가 있어서 도메인 관점에서 보면 이동하는게 좋을것같습니다

dongryeol-kim replied:
Aritcle Controller로 통합하였습니다.

ccommit-dev commented on src/main/java/com/commit/collaboration_board_server/controller/ArticleCategoryController.java:
```diff
@@ -0,0 +1,21 @@
+    private final ArticleCategoryService categoryService;
+
+    @PostMapping
+    public ResponseEntity<?> createCategory(@RequestBody ArticleCategory category) {
```
ArticleCategory 명칭이 아래 객체중 무엇인지 불분명함 
- DTO
- Request
- DAO
CreateArtricleCategoryRequest 으로 변경

SOLID 객체지향 5원칙에 S 원칙 위배시 단점을 찾아서 댓글로 작성

dongryeol-kim replied:
단일 책임 원칙을 의미하는데 하나의 클래스는 하나의 책임을 가져야 합니다. 하지만 이를 위반할 경우 한 기능이 다른 기능에 영향을 끼쳐 개발자가 원하지 않은 동작을 유발시킬 수 있습니다. 그리고 유지보수 측면에서도 코드의 가독성이 떨어질 수 있습니다.

ccommit-dev commented on src/main/java/com/commit/collaboration_board_server/controller/CommentController.java:
```diff
@@ -0,0 +1,29 @@
+    public ResponseEntity<List<Comment>> getCommentsByArticle(@PathVariable int articleId) {
+        List<Comment> comments = commentService.getNestedCommentsByArticleId(articleId);
+        return ResponseEntity.ok(comments);
+    }
```
Comment 도 Aricle 도메인 하위에 있어야 rest-ful url 규칙에 맞게 설계할수있습니다
https:///collaboration-board-server/commets/1 --> 위배
수정 해주세요

ccommit-dev commented on src/main/resources/mapper/VacationMapper.xml:
```diff
@@ -17,10 +17,10 @@
 
     <!-- UPDATE 휴가 상태 변경 -->
-    <update id="updateVacationStatus">
+    <update id="updateVacationStatus" parameterType="map">
```
map으로 써야하는 이유가있을까요?

ccommit-dev commented on src/main/java/com/commit/collaboration_board_server/controller/VacationController.java:
```diff
@@ -0,0 +1,49 @@
+import java.util.List;
+
+@RestController
+@RequestMapping("/vacations")
```
휴가등록은 유저가 하는거라서서 UserController로 옮겨주시고
휴가날짜 계산하여 violate하는부부분은 scheduleController로 옮겨주세요

ccommit-dev commented on src/main/java/com/commit/collaboration_board_server/controller/VacationController.java:
```diff
@@ -0,0 +1,49 @@
+    }
+
+    // 4. 유저: 내 휴가 목록 조회
+    @GetMapping("/my/{userNo}")
```
userNo가 이미있는데 my 의미는 중복되는거 같아서 삭제

ccommit-dev commented on src/main/java/com/commit/collaboration_board_server/service/VacationService.java:
```diff
@@ -0,0 +1,46 @@
+
+    public void applyVacation(Vacation vacation) {
+        // startTime ~ endTime 일수 계산 (휴가 일수 확인용, DB 반영은 안 함)
+        long days = ChronoUnit.DAYS.between(vacation.getStartTime(), vacation.getEndTime()) + 1;
```
휴가일수를 사용안해서 삭제

dongryeol-kim replied:
삭제했습니다.

ccommit-dev commented on src/main/java/com/commit/collaboration_board_server/service/VacationService.java:
```diff
@@ -0,0 +1,46 @@
+
+    public void approveVacation(int vacationId) {
+        Vacation vacation = vacationMapper.selectVacationById(vacationId);
+        int days = (int) ChronoUnit.DAYS.between(vacation.getStartTime(), vacation.getEndTime()) + 1;
```
기획서에 있는 정수형 휴가날짜 형태로 기능수정

dongryeol-kim replied:
수정했습니다.

ccommit-dev commented on src/main/java/com/commit/collaboration_board_server/service/VacationService.java:
```diff
@@ -0,0 +1,46 @@
+        Vacation vacation = vacationMapper.selectVacationById(vacationId);
+        int days = (int) ChronoUnit.DAYS.between(vacation.getStartTime(), vacation.getEndTime()) + 1;
+
+        vacationMapper.updateVacationStatus(vacationId, "APPROVED");
```
Enum문으로 휴가 상태 변수 수정

dongryeol-kim replied:
Entity 객체에 타입자체를 Enum형으로 변환 필요

ccommit-dev commented on src/main/java/com/commit/collaboration_board_server/service/VacationService.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    public void rejectVacation(int vacationId) {
+        vacationMapper.updateVacationStatus(vacationId, "REJECTED");
```
Enum문으로 휴가 상태 변수 수정

dongryeol-kim commented on src/main/java/com/commit/collaboration_board_server/model/Comment.java:
```diff
@@ -0,0 +1,19 @@
+    private Integer commentId; // 부모 댓글 ID
+    private Integer articleId;
+
+    private List<Comment> replies = new ArrayList<>(); // 대댓글 목록
```
안쓰고있습니다 삭제하겠습니다

dongryeol-kim committed:
issue9
 - ArticleCategory -> Article 병합
 - Vacation db 연동 구현
 - vacation 승인 및 거부 기능 추가

dongryeol-kim commented on src/main/resources/mapper/VacationMapper.xml:
```diff
@@ -0,0 +1,43 @@
+    <!-- UPDATE 유저의 남은 휴가 일수 차감 -->
+    <update id="decrementVacationCount">
+        UPDATE user
+        SET vacationCount = vacationCount - #{days}
```
수정했습니다.

dongryeol-kim committed:
issue9 코드 리팩토링
 - User api 단건/다건 api 통합
 - mapper sql 코드 가독성 포맷 수정
 - 회원 가입시 로그인 aop 적용부분 삭제
 - 불필요한 주석삭제
 - entity와 dto>request class 분리

dongryeol-kim committed:
issue9 코드 리팩토링
 - User api 단건/다건 api 통합
 - mapper sql 코드 가독성 포맷 수정
 - 회원 가입시 로그인 aop 적용부분 삭제
 - 불필요한 주석삭제
 - entity와 dto>request class 분리

dongryeol-kim committed:
issue9 코드 리팩토링
 - User api 단건/다건 api 통합
 - mapper sql 코드 가독성 포맷 수정
 - 회원 가입시 로그인 aop 적용부분 삭제
 - 불필요한 주석삭제
 - entity와 dto>request class 분리
 - xml 코드 규칙에 통일 시켜 정리

dongryeol-kim committed:
issue9 코드 리팩토링
 - User api 단건/다건 api 통합
 - mapper sql 코드 가독성 포맷 수정
 - 회원 가입시 로그인 aop 적용부분 삭제
 - 불필요한 주석삭제
 - entity와 dto>request class 분리
 - xml 코드 규칙에 통일 시켜 정리

dongryeol-kim commented on src/main/java/com/commit/collaboration_board_server/service/VacationService.java:
```diff
@@ -0,0 +1,46 @@
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
```
이 어노테이션은 초기화 되지 않은 final필드나, @NonNull이 붙은 필드에 대해 생성자를 생성해 주는 역할을 합니다.

dongryeol-kim committed:
issue9 코드 리팩토링
 - mysql query format pretty

dongryeol-kim committed:
issue9 코드 리팩토링
 - response 리팩토링

dongryeol-kim committed:
issue9 코드 리팩토링
 - 예외처리 및 유효성 검사
